controllers/templatesync/template_sync.go

```diff
@@ -518,12 +518,21 @@ func (r *PolicyReconciler) processDependencies(ctx context.Context, dClient dyna
 			return nil, err
 		}
 
-		// set up namespace for replicated policy dependencies
 		ns := dep.Namespace
+
+		// infer namespace if not provided
 		if ns == "" {
 			ns = r.ClusterNamespace
 		}
 
+		// override given namespace if one of our policy types
+		// (these will always be in the cluster namespace)
+		if dep.Group == policiesv1.GroupVersion.Group &&
+			dep.Version == policiesv1.GroupVersion.Version &&
+			strings.HasSuffix(dep.Kind, "Policy") {
+			ns = r.ClusterNamespace
```

Comment:
I'm trying to decide whether this should be an error case? It seems like we don't necessarily want to assume that the user should be overridden, but maybe I'm missing some implementation details here.

Reply:
It could be an error. I was trying to go for ease-of-use here.

Example: I want my policy to be active when a separate ConfigurationPolicy is Compliant. What namespace should I use for the dependency? I might think that I should use the namespace of the root Policy containing the ConfigurationPolicy, but actually that namespace only exists on the hub. On the managed cluster, the ConfigurationPolicy exists in the managed cluster namespace, but I can't put that in the dependency definition because each cluster is different.

So this makes the framework always use the cluster namespace, instead of requiring the user to leave the namespace blank.

Reply:
It makes sense for ease of use, but I suppose I don't like that it could propagate misconceptions if the user puts in a namespace they think is correct and we override it without letting them know it's actually wrong. I think I'd prefer an informative message that they need to remove or update the namespace.

Reply:
Yeah, "propagating misconceptions" definitely sounds like a bad thing. You've convinced me! I'll look at putting a warning/error in the propagator logic for this situation instead of overriding it here.